arduino-ide-extension/src/browser/contributions/init-libs-platforms.ts

```diff
@@ -17,35 +17,81 @@ export class InitLibsPlatforms extends Contribution {
       InitLibsPlatforms.INIT_LIBS_AND_PACKAGES
     ));
     if (isFirstStartup) {
-      await this.localStorageService.setData(
-        InitLibsPlatforms.INIT_LIBS_AND_PACKAGES,
-        true
-      );
       const avrPackage = await this.boardsService.getBoardPackage({
         id: 'arduino:avr',
       });
       const builtInLibrary = (
-        await this.libraryService.search({
-          query: 'Arduino_BuiltIn',
-        })
+        await this.libraryService.search({ query: 'Arduino_BuiltIn' })
       )[0];
 
+      let avrPackageError: Error | undefined;
+      let builtInLibraryError: Error | undefined;
+
       if (avrPackage) {
         try {
           await this.boardsService.install({
             item: avrPackage,
             noOverwrite: true, // We don't want to automatically replace custom platforms the user might already have in place
           });
-        } catch {} // If this fails, we still want to install the libraries
+        } catch (e) {
+          // There's no error code, I need to parse the error message: https://github.com/arduino/arduino-cli/commit/ffe4232b359fcfa87238d68acf1c3b64a1621f14#diff-10ffbdde46838dd9caa881fd1f2a5326a49f8061f6cfd7c9d430b4875a6b6895R62
+          if (
+            e.message.includes(
+              `Platform ${avrPackage.id}@${avrPackage.installedVersion} already installed`
+            )
+          ) {
+            // If arduino:avr installation fails because it's already installed we don't want to retry on next start-up
+            console.error(e);
+          } else {
+            // But if there is any other error (e.g.: no interntet cconnection), we want to retry next time
+            avrPackageError = e;
+          }
+        }
+      } else {
+        avrPackageError = new Error('Could not find platform.');
       }
+
       if (builtInLibrary) {
         try {
           await this.libraryService.install({
             item: builtInLibrary,
             installDependencies: true,
             noOverwrite: true, // We don't want to automatically replace custom libraries the user might already have in place
           });
-        } catch {}
+        } catch (e) {
+          // There's no error code, I need to parse the error message: https://github.com/arduino/arduino-cli/commit/2ea3608453b17b1157f8a1dc892af2e13e40f4f0#diff-1de7569144d4e260f8dde0e0d00a4e2a218c57966d583da1687a70d518986649R95
+          if (
+            e.message.includes(
+              `Library ${builtInLibrary.name} is already installed`
```

Comment:
This does not catch failures caused by a prevention of overwriting [a dependency](https://github.com/arduino-libraries/Arduino_BuiltIn/blob/1.0.0/library.properties#L10) of the "Arduino_BuiltIn" library.

The error produced in that case does not match against the pattern. For example:

```text
Library SD@1.2.4 is already installed, but with a different version: SD@1.2.3
```

### To reproduce

1. Install an outdated version of one of the dependencies of the ["**Arduino_BuiltIn**"](https://github.com/arduino-libraries/Arduino_BuiltIn) library (e.g., ["**SD**"](https://github.com/arduino-libraries/SD) version 1.2.3).
1. Select **File > Quit** from the Arduino IDE menus if it is running
1. Rename or delete (:warning: cautiously) the following folder to simulate a first run:
   - Windows:
     ```
     C:\Users\<user name>\AppData\Roaming\arduino-ide
     ```
   - Linux:
     ```
     ~/.config/arduino-ide
     ```
   - macOS:
     ```
     ~/Library/Application Support/arduino-ide
     ```
1. Start the Arduino IDE.
1. Wait for the conclusion of the first run installations.
   🙂 Overwrite of the dependency library is prevented as expected:
   ```text
   Error: 2 UNKNOWN: Library SD@1.2.4 is already installed, but with a different version: SD@1.2.3
   ```
1. Select **File > Quit** from the Arduino IDE menus.
1. Start the Arduino IDE.

🐛 The first run installations are attempted again, failing as before due to the SD library overwrite prevention. This will occur every time the IDE is started.

### Expected behavior

First run installations are not attempted again if they failed due to the intended overwrite prevention behavior.